pkg/operators/otel-metrics/otel-metrics.go

```diff
@@ -374,6 +398,50 @@ func (mc *metricsCollector) addValHistFunc(f datasource.FieldAccessor) error {
 	switch f.Type() {
 	default:
 		return fmt.Errorf("unsupported field type for metrics value %q: %s", f.Name(), f.Type())
+	case api.ArrayOf(api.Kind_Uint32), api.ArrayOf(api.Kind_Uint64):
+		// Calc buckets
+		typeLen := 4
+		extractor := func(b []byte) uint64 { return uint64(ds.ByteOrder().Uint32(b)) }
+		if f.Type() == api.ArrayOf(api.Kind_Uint64) {
+			typeLen = 8
+			extractor = ds.ByteOrder().Uint64
+		}
+		alen := int(f.Size()) / typeLen
+
+		bucketVals := make([]int64, alen)
+		bucketValsFloat := make([]float64, alen)
+		for i := range bucketVals {
+			bucketVals[i] = int64(1 << i)
+			bucketValsFloat[i] = float64(bucketVals[i])
+		}
+
+		options = append(options, metric.WithExplicitBucketBoundaries(bucketValsFloat...))
+		hOptions := make([]metric.Int64HistogramOption, len(options))
+		for i, option := range options {
+			hOptions[i] = option
+		}
+
+		hist, err := mc.meter.Int64Histogram(f.Name(), hOptions...)
+		if err != nil {
+			return fmt.Errorf("adding metric histogram for %q: %w", f.Name(), err)
+		}
+		mc.values = append(mc.values, func(ctx context.Context, data datasource.Data, set attribute.Set) {
+			b := f.Get(data)
+			if len(b)%typeLen != 0 {
+				return
```

Comment:
I don't like logging here - that's a hot path and when there are thousands of events coming in, that would lead to excessive logging. It's a super rare case IMHO (you need to mess up somewhere else) and the check is only there to not crash.